aws/data_source_aws_qldb_ledger_test.go

```diff
@@ -0,0 +1,52 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceAwsQLDBLedger(t *testing.T) {
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(7)) // QLDB name cannot be longer than 32 characters
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsQLDBLedgerConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.aws_qldb_ledger.by_name", "arn", "aws_qldb_ledger.tf_test", "arn"),
+					resource.TestCheckResourceAttrPair("data.aws_qldb_ledger.by_name", "deletion_protection", "aws_qldb_ledger.tf_test", "deletion_protection"),
+					resource.TestCheckResourceAttrPair("data.aws_qldb_ledger.by_name", "name", "aws_qldb_ledger.tf_test", "name"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAwsQLDBLedgerConfig(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_qldb_ledger" "tf_wrong1" {
+  name = "%[1]s1"
```

Comment:
Since each of these three QLDB Ledgers in the test configuration have the same name, this testing is currently failing:

```
--- FAIL: TestAccDataSourceAwsQLDBLedger (34.19s)
    testing.go:615: Step 0 error: errors during apply:
        
        Error: Error creating QLDB Ledger: ResourceAlreadyExistsException: Ledger name already exists
```

Changing the name arguments to end in 1, 2, and 3 would resolve this issue.